### PR TITLE
Update vectormath

### PR DESCRIPTION
**Pull request by miigotu**
Signed-off-by: miigotu <miigotu@gmail.com>

### CODE REVIEW

**Comment by miigotu** [comment]
@bucanero - The samples build now with this PR, can you maybe run them and check if vectormath is working?

**Comment by bucanero** [comment]
Sure, I'll about to check out this branch, build the binaries and if I get a .self I'll try it on a real PS3.

**Comment by bucanero** [comment]
Alright, here's my report:

Even after merging this PR into my local copy, I was still getting the `narrowing conversion` error when trying to build the `rsxtest` example.
Then I tried adding the #75 patch, and that brought me the gcc `internal compiler error`.

Based on feedback from another dev, the compiler issue was due to a `-Os` compiler flag. 

So I removed `-Os` from the Makefile, and I was able to build the example.

Finally, I tested `rsxtest.self` on my PS3 and it worked like a charm.

Just to be sure, I also tested with the original `vectormath` code (master branch), applying only the #75 patch, and removing the `-Os` gave me positive results. I was able to build the sample, and run the binary .self on my PS3.

Long story short, I checked that only #75 and a change on the `Makefile` is needed. 

Hopefully my tests can help you decide what stuff to merge or not. :)

**Comment by miigotu** [comment]
This branch isn't able to use the same patch as #70 so I'm a bit confused here.

I will build the rsx example with this branch and post the self later, if you can just check if that runs that would be cool.

Ideally, using the updated vectormath is better than patching the older vectormath.

**Comment by bucanero** [comment]
Well what I actually did was applying the patch manually:

I edited the file and changed the cast to vec_uint4 so I didn’t get the narrowing conversion error. 

I hope that explains it better. 😄

For sure, if you share the self I’ll test it on my PlayStation 3. 

At least on my end, the new vector math was still giving me the narrowing error. I guess it can also be fixed adding that flag on the make file

**Comment by miigotu** [comment]
You have to rebuild PSL1GHT after, and then also make install or you still have the old headers =P
Anyways, I rebuilt it, didn't like the outcome. We will merge #75 (I also fixed it for SPU)
